File: code/238.product-of-array-except-self.php
<?php
/*
 * @lc app=leetcode id=238 lang=php
 *
 * [238] Product of Array Except Self
 */

// @lc code=start
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function productExceptSelf($nums) {
        // e.g.
        $count = count($nums);
        $result = array_fill(0, $count, 1);

        $leftProduct = 1;
        for ($i = 0; $i < $count; $i++) {
            $result[$i] *= $leftProduct;
            $leftProduct *= $nums[$i];
        }

        $rightProduct = 1;
        for ($i = $count - 1; $i >= 0; $i--) {
            $result[$i] *= $rightProduct;
            $rightProduct *= $nums[$i];
        }
        return $result;
    }
}
// @lc code=end

File: code/238.product-of-array-except-self.go
package code

/*
 * @lc app=leetcode id=238 lang=golang
 *
 * [238] Product of Array Except Self
 */

// @lc code=start
func productExceptSelf(nums []int) []int {
	result := make([]int, len(nums))

	left := 1
	for i := 0; i < len(nums); i++ {
		result[i] = left
		left *= nums[i]
	}

	right := 1
	for i := len(nums) - 1; i >= 0; i-- {
		result[i] *= right
		right *= nums[i]
	}

	return result
}

// @lc code=end
